File: src/main/java/de/ayont/lpc/channels/Channel.java
package de.ayont.lpc.channels;

import de.ayont.lpc.LPC;
import net.kyori.adventure.text.Component;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.Nullable;

import java.util.Set;
import java.util.UUID;

public interface Channel {

    String getId();

    String getName();

    String getFormat();

    String getPermission();

    String getShortcut();

    String getSymbol();

    boolean canJoin(Player player);

    boolean canRead(Player player, @Nullable Player sender);

    void sendMessage(Player sender, String message);
    
    // For when the player just switches to the channel, send a confirmation message?
    void onJoin(Player player);
}

File: src/main/kotlin/de/ayont/lpc/channels/Channel.kt
package de.ayont.lpc.channels

import org.bukkit.entity.Player

interface Channel {

    fun getId(): String

    fun getName(): String

    fun getFormat(): String

    fun getPermission(): String

    fun getShortcut(): String

    fun getSymbol(): String

    fun canJoin(player: Player): Boolean

    fun canRead(player: Player, sender: Player): Boolean

    fun sendMessage(sender: Player, message: String)
    
    // For when the player just switches to the channel, send a confirmation message?
    fun onJoin(player: Player)
}
